sqltap/sqltap.py

```diff
@@ -296,6 +338,8 @@ def __init__(self, stats, report_file=None, report_dir=".",
 
         :param template_dir: folder of the template to generate the report.
         """
+        self.duration = ((stats[-1].end_time - stats[0].start_time)
+                         if stats else 0)
```

Comment:
You're entirely right, I missed the falsey ternary check. Sorry!

In order to do the proper time-based reporting we'd need to modify the `report()` API to take additional parameters :( I think we'll skip that because it's an API breaking change. I'll open an issue to track it though.